src/main/groovy/lang/ObjectRange.java

```diff
@@ -488,37 +500,61 @@ public void remove() {
 
     @Override
     public List step(int step) {
+        return step((Number) step);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List step(Number step) {
         final IteratorClosureAdapter adapter = new IteratorClosureAdapter(this);
         step(step, adapter);
         return adapter.asList();
     }
 
     /**
-     * Increments by one
+     * Increments by step size
      *
      * @param value the value to increment
-     * @return the incremented value
+     * @param step  the value to increment by
+     * @return the incremented value or null, if there isn't any
      */
-    private static Object increment(Object value) {
+    protected static Comparable increment(Comparable value, Number step) {
         if (value instanceof Number) {
-            return plus((Number) value, 1);
-        } else {
-            return InvokerHelper.invokeMethod(value, "next", null);
+            return (Comparable) plus((Number) value, step);
+        }
+
+        for (int i = 0; compareLessThan(i, step); i++) {
```

Comment:
Could you please provide alternatives also, when writing criticism, otherwise it seems just nitpicking.

I consider that the advantages outweigh the above mentioned problems for the above mentioned reasons.